### PR TITLE
Fix export types for typescript client (#6805)

### PR DESCRIPTION
**Pull request by Drago92**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by joscha** [comment]
@macjohnny Looking at 598b5097afc89090e43f597bc7a1a5fb7a9c6136 I think this would actually make sense. It would help any post-processing of the TS when bundling, etc. - is there a reason we only do this for Deno?

**Comment by joscha** [comment]
relates to #19484

File: docs/generators/fsharp-functions.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |prependFormOrBodyParameters|Add form or body parameters to the beginning of the parameter list.| |false|
 |sortModelPropertiesByRequiredFlag|Sort model properties to place required parameters before optional parameters.| |true|
 |sortParamsByRequiredFlag|Sort method arguments to place required parameters before optional parameters.| |true|
-|sourceFolder|source folder for generated code| |OpenAPI/src|
+|sourceFolder|source folder for generated code| |OpenAPI\src|
```
Comment (macjohnny) [inline docs/generators/fsharp-functions.md:37]: please manually revert to `/`, as this is the standard for linux-based systems

**Review by macjohnny — CHANGES_REQUESTED** [review]
thanks for your contribution

**Comment by macjohnny** [comment]
> @macjohnny Looking at [598b509](https://github.com/OpenAPITools/openapi-generator/commit/598b5097afc89090e43f597bc7a1a5fb7a9c6136) I think this would actually make sense. It would help any post-processing of the TS when bundling, etc. - is there a reason we only do this for Deno?

IIRC this was because it was introduced with TypeScript 3.8 https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html, so to have a lower version requirement for non-deno code?

**Comment by macjohnny** [comment]
> That would make sense. What is the guidance for our support of TS versions? 3.8 was released on 20 February 2020. 4+ years is a long time ;)

yeah, looking at https://github.com/OpenAPITools/openapi-generator/blob/df22ca6dccfdaf93b3bcec71953899478d2ce35d/modules/openapi-generator/src/main/resources/typescript/package.mustache#L78, it seems we now require at least typescript 4.0

**Comment by joscha** [comment]
> > That would make sense. What is the guidance for our support of TS versions? 3.8 was released on 20 February 2020. 4+ years is a long time ;)
> 
> yeah, looking at
> 
> https://github.com/OpenAPITools/openapi-generator/blob/df22ca6dccfdaf93b3bcec71953899478d2ce35d/modules/openapi-generator/src/main/resources/typescript/package.mustache#L78
> 
> , it seems we now require at least typescript 4.0

Blame says 3.x support was dropped about 2 years ago already in 8c57b66da90bdf1d5a9a7d3588458731f0dec078, so I think this pull request is a go?

**Comment by macjohnny** [comment]
> Blame says 3.x support was dropped about 2 years ago already in [8c57b66](https://github.com/OpenAPITools/openapi-generator/commit/8c57b66da90bdf1d5a9a7d3588458731f0dec078), so I think this pull request is a go?

yes, after the paths are fixed we can merge

File: samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES [hunk 2]
```diff
@@ -5,6 +5,7 @@
 .travis.yml
 DESCRIPTION
 NAMESPACE
+README.md
```
Comment (joscha) [inline samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES:8]: seems like the sort ordering is not kept stable across systems for this one

**Comment by joscha** [comment]
https://github.com/OpenAPITools/openapi-generator/pull/19725 has the changes and commits from here + latest master + regenerated samples.

File: samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES (hunk 2 above)
Comment (macjohnny) [reply samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES:8]: yes, probably because the samples were generated on a windows machine

Comment (macjohnny) [reply samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES:8]: can you fix it?

Comment (joscha) [reply samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES:8]: trying already. Always tricky to do non-platform specific things :)

Comment (joscha) [reply samples/client/petstore/R-httr2-wrapper/.openapi-generator/FILES:8]: https://github.com/OpenAPITools/openapi-generator/pull/19726

**Comment by joscha** [comment]
@Drago92 if you have a second, it would be amazing if you could regenerate in https://github.com/OpenAPITools/openapi-generator/pull/19725 on your system via:
```
gh pr checkout 19725
```
and then
```
./mvnw clean package 
./bin/generate-samples.sh ./bin/configs/*.yaml
./bin/utils/export_docs_generators.sh
```
to see if the fix works as expected.

**Comment by Drago92** [comment]
> @Drago92 if you have a second, it would be amazing if you could regenerate in #19726 on your system via:
> 
> ```
> gh pr checkout 19726
> ```
> 
> and then
> 
> ```
> ./mvnw clean package 
> ./bin/generate-samples.sh ./bin/configs/*.yaml
> ./bin/utils/export_docs_generators.sh
> ```
> 
> to see if the fix works as expected.

Your fix works and the slashes are correct now. But the line Endings will be converted on Windows:
image
image

and in .dist-Files a "\" will be added after src:
image

**Comment by joscha** [comment]
Thank you for checking @Drago92  - I didn't fix the first or last image, that is a separate codepath. I didn't touch the line-endings, they are `\n` currently, probably if you saved the file they will become `\r\n`. We could possibly add a `.gitattributes` file with:

```gitattributes
FILES text eol=lf
```
which should then keep them as `\n`. You could try that if you wanted?

**Comment by Drago92** [comment]
> Thank you for checking @Drago92 - I didn't fix the first or last image, that is a separate codepath. I didn't touch the line-endings, they are `\n` currently, probably if you saved the file they will become `\r\n`. We could possibly add a `.gitattributes` file with:
> 
> ```gitattributes
> FILES text eol=lf
> ```
> 
> which should then keep them as `\n`. You could try that if you wanted?

Yeah it will help :)

**Pull request closed without merging**